plugins/modules/haproxy.py

```diff
@@ -343,7 +343,7 @@ def execute_for_backends(self, cmd, pxname, svname, wait_for_status=None):
 
             if state is not None:
                 self.execute(Template(cmd).substitute(pxname=backend, svname=svname))
-                if self.wait:
+                if self.wait and wait_for_status != "DRAIN" and state != "DOWN":
```

Comment:
Shouldn't this be
```suggestion
                if self.wait and (wait_for_status != "DRAIN" or state != "DOWN"):
```
then? Or equivalently:
```suggestion
                if self.wait and not (wait_for_status == "DRAIN" and state == "DOWN"):
```
Otherwise it will **never** wait if `wait_for_status == 'DRAIN'` (no matter what `state` is), or if `state == 'DOWN'` (no matter what `wait_for_status` is).

Reply:
I don't really see any improvement here. The first suggestion is crap, as you write yourself. The second suggestion is just a different codestyle.

Reply:
I strongly disagree with that my first suggestion is crap. I did not write that. I mainly provided two different ways of writing the same form (that differ by an application of [De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) so you can pick the version you find easier to understand), which are **not** equivalent to your expression. Your expression covers way too many situations that it should not. My expressions mitigate that and stick to the specific situation you mentioned in https://github.com/ansible-collections/community.general/pull/8100#issuecomment-2003246133.

Your expression is equivalent to `self.wait and not (wait_for_status == "DRAIN" or state == "DOWN")` (again by De Morgan's laws).

Reply:
Ok, you are right. I'm stupid.